container/libcontainer/handler.go

```diff
@@ -803,9 +803,13 @@ func setCPUStats(s *cgroups.Stats, ret *info.ContainerStats, withPerCPU bool) {
 	}
 	numActual = minUint32(numPossible, numActual)
 	ret.Cpu.Usage.PerCpu = make([]uint64, numActual)
+	ret.Cpu.Usage.PerCpuKernel = make([]uint64, numActual)
+	ret.Cpu.Usage.PerCpuUser = make([]uint64, numActual)
 
 	for i := uint32(0); i < numActual; i++ {
 		ret.Cpu.Usage.PerCpu[i] = s.CpuStats.CpuUsage.PercpuUsage[i]
+		ret.Cpu.Usage.PerCpuKernel[i] = s.CpuStats.CpuUsage.PercpuUsageInKernelmode[i]
```

Comment:
lenght of elements in `s.CpuStats.CpuUsage.PercpuUsageInKernelmode` and in `s.CpuStats.CpuUsage.PercpuUsageInUsermode` is not checked, imho it is not safe.

Reply:
Is it possible though? Both percpu and percpu in user/kernel come from cgroups. percpu from cpuacct.usage_percpu and per mode is computed from cpuacct.usage_all

Reply:
I think that it is worth to add unit test for case when `len(s.CpuStats.CpuUsage.PercpuUsage) !=0` and `len( s.CpuStats.CpuUsage.PercpuUsageInKernelmode) == len(s.CpuStats.CpuUsage.PercpuUsageInUsermode)  == 0`. What do you think?